### ISSUE TITLE
DATE_PHRASES are  not displayed correctly

### ISSUE THREAD

**Issue by webtrees-pesz** [fisharebest/webtrees#5011]
1 BIRT
2 DATE INT 1900 (1. January 1900)
is displayed but

1 BIRT
2 DATE (1. January 1900) 
is **not** displayed.

GEDCOM-5.5.1 says:
DATE_VALUE:= {Size=1:35}
[
<DATE> |
<DATE_PERIOD> |
<DATE_RANGE>|
<DATE_APPROXIMATED> |
INT <DATE> (<DATE_PHRASE>) |
**(<DATE_PHRASE>)**
]

**Comment by arbor95** [comment]
should be fixed with pull request #5012

**Comment by fisharebest** [comment]
`$match[1]` can contain user-generated text.  So it needs to be escaped using `e()`.

Would it be better to put this fix in `Date::display()` so that it works everywhere?

e.g. at https://github.com/fisharebest/webtrees/blob/main/app/Date.php#L181-L182 add something like

```
if ($this->text !== '') {
    $tmp .= ' (' . e($text) . ')';
}
```

(Note: I haven't tested this.  It's just a quick idea from looking at the code).

**Comment by webtrees-pesz** [comment]
Thank you Franz and Greg for your quick support.

 

Best regards

Peter

 

Von: Franz Frese ***@***.***> 
Gesendet: Freitag, 26. Juli 2024 14:16
An: fisharebest/webtrees ***@***.***>
Cc: webtrees-pesz ***@***.***>; Author ***@***.***>
Betreff: Re: [fisharebest/webtrees] DATE_PHRASES are not displayed correctly (Issue #5011)

 

should be fixed with pull request #5012 <https://github.com/fisharebest/webtrees/pull/5012>

**Comment by arbor95** [comment]
> `$match[1]` can contain user-generated text. So it needs to be escaped using `e()`.
> 
> Would it be better to put this fix in `Date::display()` so that it works everywhere?
> 
> e.g. at https://github.com/fisharebest/webtrees/blob/main/app/Date.php#L181-L182 add something like
> 
> ```
> if ($this->text !== '') {
>     $tmp .= ' (' . e($this->text) . ')';
> }
> ```
> 
> (Note: I haven't tested this. It's just a quick idea from looking at the code).

Seems as if the current lines 181 and 182 are used for many (other) purposes, that I don't overview.
So I would be content, when Dates in facts are displayed corresponding to gedcom definition.
Sure with escaping (will learn. Sorry, I'm not used with php and so on).
But if you mean, its better changing Date.php, do. (a quick test with the above, did not work).
Tell me, whether I shall drop or modify the corresponding pull request.

**Comment by fisharebest** [comment]
Fixed by #5015
